core/src/main/java/io/github/tcdl/msb/collector/Collector.java

```diff
@@ -253,6 +238,30 @@ private int setResponsesRemainingForResponderId(String responderId, Integer resp
         return responsesRemainingById.get(responderId);
     }
 
+    private int getWaitForResponsesFromConfigs(RequestOptions requestOptions) {
+        if (requestOptions.getWaitForResponses() == null || requestOptions.getWaitForResponses() == WAIT_FOR_RESPONSES_UNTIL_TIMEOUT) {
+            // use for Infinity number or expected responses
+            return Integer.MAX_VALUE;
```

Comment:
Is `MAX_VALUE` large enougth? Looks like this logic might be broken if we get back more than 2147483647 responses. Why bother with that? You have a whole spectrum of negative numbers to describe this semantic (waiting for responses until timeout reached).